### DISCUSSION TITLE
Paginate returns wrong base path

### DISCUSSION THREAD

**Discussion by peirix** [laravel/framework#45924] [Q&A]
I'm having an issue with pagination on my laravel app, where it keeps returning every url with a `http://` prefix, and not a `https://`. And the strangest thing is that it seems to be _only_ for pagination urls. So if I output a url using the `url()` helper it will use the correct one.

Example of this can be seen here: https://epion.mustasj.dev/api/force

Which is set up like this:

```php
Route::get('/force', function () {
    return response()->json([
        'built_in_url' => url('/api/force'),
        'paginate' => Manufacturer::paginate(20)
    ]);
});
```

This is set up on Digital Ocean's App Platform. I've tried manually setting `APP_URL` to `https://epion.mustasj.dev`, I've tried explicitly setting `URL::forceScheme('https');` in `AppServiceProvider`, in `RouteServiceProvider` and in `web.php` in the routes directory. I've tried setting `FORCE_HTTPS=true` in the env file.

Really at a loss here, and not sure if there's something with Laravel, or maybe something with my setup at Digital Ocean. Any hints on how to dig into this deeper to find out where it's going wrong?

**Comment by denizgolbas** [comment]
```console
php artisan optimize:clear
```
Did you run this after env changes?

**Comment by peirix** [comment]
Yes. And I also ran `php artisan route:clear` just in case. And `composer dump-autoload`.

**Comment by kitro** [comment]
Same issue as above, I tried also to inject `$this->app['request']?->server?->set('HTTPS', true);` in the boot method in AppServiceProvider. not works in L10.

**Reply by kitro** [reply]
Resolved by this solution from  @Travis-Britz https://github.com/laravel/framework/issues/15361#issuecomment-543840548

**Reply by abalest14** [reply]
> Resolved by this solution from @Travis-Britz [#15361 (comment)](https://github.com/laravel/framework/issues/15361#issuecomment-543840548)

Hi Kitro, i have the same issue, wich kind of fix did you apply? Thanks

**Reply by Afrid1719** [reply]
`$this->app['request']?->server?->set('HTTPS', true);`
It worked for me, thanks though.
